clippy_lints/src/doc/doc_comment_double_space_linebreak.rs

```diff
@@ -0,0 +1,52 @@
+use clippy_utils::diagnostics::span_lint_and_then;
+use rustc_ast::token::CommentKind;
+use rustc_ast::{AttrKind, AttrStyle, Attribute};
+use rustc_errors::Applicability;
+use rustc_lint::LateContext;
+use rustc_span::Span;
+
+use super::DOC_COMMENT_DOUBLE_SPACE_LINEBREAK;
+
+pub fn check(cx: &LateContext<'_>, attrs: &[Attribute]) {
+    let replacements: Vec<_> = collect_doc_replacements(attrs);
+
+    if let Some((&(lo_span, _), &(hi_span, _))) = replacements.first().zip(replacements.last()) {
+        span_lint_and_then(
+            cx,
+            DOC_COMMENT_DOUBLE_SPACE_LINEBREAK,
+            lo_span.to(hi_span),
+            "doc comments should use a back-slash (\\) instead of a double space to indicate a linebreak",
+            |diag| {
+                diag.multipart_suggestion(
+                    "replace this double space with a back-slash",
+                    replacements,
+                    Applicability::MachineApplicable,
+                );
+            },
+        );
+    }
+}
+
+fn collect_doc_replacements(attrs: &[Attribute]) -> Vec<(Span, String)> {
+    attrs
+        .iter()
+        .filter_map(|attr| {
+            if let AttrKind::DocComment(com_kind, sym) = attr.kind
+                && com_kind == CommentKind::Line
+                && let comment = sym.as_str()
+                && comment.ends_with("  ")
```

Comment:
Somewhere in this lint, we should also check that the item or doc attributes don't come from macro expansion. I think this iterator might be the right place. A simple `!attr.span.from_expansion()` should do the trick.

It would also be cool if you could craft a test example.